### PR TITLE
[Merged by Bors] - Move to smallvec v1.6

### PR DESCRIPTION
**Pull request by TheRawMeatball**

### CODE REVIEW

**Comment by TheRawMeatball** [comment]
Since we're already using latest stable, should be enable the `union` and `const_generics` features?

**Comment by mockersf** [comment]
yes for `union`!

no for `const_generics`: https://docs.rs/smallvec/1.6.1/smallvec/#const_generics, unstable / needs nightly according to their code

**Comment by TheRawMeatball** [comment]
The docs aren't up-to-date: in 1.6, it compiles on stable :)

**Comment by cart** [comment]
Ooh yeah lets enables those features :)

**Comment by TheRawMeatball** [comment]
Hmm, what would be a good place to enable these features?

I'm adding a `smallvec` dependency to `bevy_ecs` in #2071, maybe that's a good place to add it? Or should we add it in all places that have a dependency already?

**Comment by cart** [comment]
Hmm I think it doesn't matter too much as features are merged across dependencies. Keeping them in sync makes it easier to do a simple "search for `smallvec =` globally and ensure everything lines up", but it also means a crate is opting into features it might not need.

Does smallvec respect the soft "features are additive only" rule or will enabling something like const_generics break apis that weren't coded to it?

We could also consider re-exporting smallvec from `bevy_utils`, but I'm not sure I want that to become a dumping ground for every shared dependency with custom features.

**Comment by TheRawMeatball** [comment]
> Does smallvec respect the soft "features are additive only" rule or will enabling something like const_generics break apis that weren't coded to it?

From what I can see, `union` is an impl-level optimization that shouldn't break anything, and enabling const generics simply adds more impls.

**Comment by mockersf** [comment]
> We could also consider re-exporting smallvec from `bevy_utils`, but I'm not sure I want that to become a dumping ground for every shared dependency with custom features.

I don't think that's necessary, those features will be enabled if someone import `smallvec` in their project even if they don't enable them themselves

File: crates/bevy_render/Cargo.toml [hunk 1]
```diff
@@ -31,9 +31,8 @@ image = { version = "0.23.12", default-features = false }
 # misc
 serde = { version = "1", features = ["derive"] }
 bitflags = "1.2.1"
-smallvec = "1.4.2"
-# TODO: replace once_cell with std equivalent if/when this lands: https://github.com/rust-lang/rfcs/pull/2788
-once_cell = "1.4.1"
+smallvec = { version = "1.6", feautres = ["union", "const_generics"] }
```
Comment (y15un) [inline crates/bevy_render/Cargo.toml:34]: typo found: `feautres` => `features`.

cargo is complaining about this "unused manifest key."

File: crates/bevy_ui/Cargo.toml [hunk 2]
```diff
@@ -33,4 +33,4 @@ bevy_utils = { path = "../bevy_utils", version = "0.5.0" }
 # other
 stretch = "0.3"
 serde = {version = "1", features = ["derive"]}
-smallvec = "1.4"
\ No newline at end of file
+smallvec = { version = "1.6", feautres = ["union", "const_generics"] }
```
Comment (y15un) [inline crates/bevy_ui/Cargo.toml:36]: typo found: `feautres` => `features`.

cargo is complaining about this "unused manifest key."

**Review by y15un — COMMENTED** [review]
hi, i found these two instances of simple typo; cargo was complaining while I was trying to run examples.

File: crates/bevy_render/Cargo.toml (hunk 1 above)
Comment (mockersf) [reply crates/bevy_render/Cargo.toml:34]: could be fixed in https://github.com/bevyengine/bevy/pull/2111

**Comment by mockersf** [comment]
> hi, i found these two instances of simple typo; cargo was complaining while I was trying to run examples.

they could be fixed in https://github.com/bevyengine/bevy/pull/2111

**Pull request closed without merging**